src/duplicacy_storage.go

```diff
@@ -461,6 +461,38 @@ func CreateStorage(preference Preference, resetPassword bool, threads int) (stor
 		SavePassword(preference, "s3_secret", secretKey)
 
 		return storage
+
+	} else if matched[1] == "wasabi" {
+
+		region := matched[2]
+		endpoint := matched[3]
+		bucket := matched[5]
+
```

Comment:
The s3 storage strips off the trailing @ from the region.  We should do the same here.

Reply:
Ah, I see what's going on: the root of the problem is the [regexp that picks the URL apart](https://github.com/gilbertchen/duplicacy/blob/e07226bd621dcf03b15193e60ecbcb9efd3e797c/src/duplicacy_storage.go#L264) collecting the `@` and anything afterward consuming it has to compensate.

I've adjusted the Wasabi code to make the correction so it's at least consistent with everything else.

Better would be to correct the problem at the root and not collect the `@` in the first place so it doesn't have to be band-aided later.  Better still would be to use `net/url` to parse the URL.  Both would require modifications outside the scope of this PR.